tests: bsim: set log level from command-line argument

00thirdeye00 opened the pull request:
Implementation to take command-line arguments to set log level in run time for bsim module

Fixes: #75249

theob-pro commented:
Hi @00thirdeye00, thanks for the PR!

Looking at your discussion with @jori-nordic it seems that you may have misunderstood him. When he was describing `-argtest log_level 4` it was just an example on how the final interface may look like. But as he describe in his message, you need to first implement the logic for that `log_level` argument. So, as said by jori-nordic, that goes by:

1. Registering a function that will parse the argument you're adding;
2. Set global variable that will store the value of the arguments passed;
3. Use the maximum log level (debug) by default;
4. Enable runtime filtering of logs (with [`LOG_RUNTIME_FILTERING`](https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_LOG_RUNTIME_FILTERING));
5. Set the log level according to the global variable previously set with either the testlib as linked by jori-nordic or with the logging API [`log_filter_set`](https://docs.zephyrproject.org/latest/doxygen/html/group__log__ctrl.html#ga32956e4ba4ed92e5e5aeb5503be0047e).

If you have any questions, feel free to ask :slightly_smiling_face:

Thalley commented on tests/bsim/bluetooth/host/misc/sample_test/src/main.c:
```diff
@@ -9,11 +9,33 @@
 extern void entrypoint_peer(void);
 extern enum bst_result_t bst_result;
 
+static uint8_t	set_log_level_run_time = 3;
+
+static void test_args(int argc, char *argv[])
+{
+	size_t argn = 0;
+	const char *arg = argv[argn];
+
+	if(strcmp(arg, "log_level") == 0){
+		set_log_level_run_time = strtoul(argv[++argn], NULL, 10);
+
+		TEST_PRINT("arg : %d\n", set_log_level_run_time);
+
+		if(set_log_level_run_time >= 0 && 
+			set_log_level_run_time <= 4){
```
Suggest to use the predefined values
```suggestion
static uint8_t set_log_level_run_time = LOG_LEVEL_INF;

static void test_args(int argc, char *argv[])
{
	size_t argn = 0;
	const char *arg = argv[argn];

	if(strcmp(arg, "log_level") == 0){
		set_log_level_run_time = strtoul(argv[++argn], NULL, 10);

		TEST_PRINT("arg : %d\n", set_log_level_run_time);

		if(set_log_level_run_time <= LOG_LEVEL_DBG){
```

00thirdeye00 replied:
changed variable initialization to predefined values

Thalley commented on tests/bsim/bluetooth/host/misc/sample_test/src/main.c:
```diff
@@ -9,11 +9,33 @@
 extern void entrypoint_peer(void);
 extern enum bst_result_t bst_result;
 
+static uint8_t	set_log_level_run_time = 3;
```
This doesn't need to be a global variable

00thirdeye00 replied:
it was my understanding from the description. i can make it local to the `test_args` function

Thalley commented on tests/bsim/bluetooth/host/misc/sample_test/src/main.c:
```diff
@@ -9,11 +9,33 @@
+	const char *arg = argv[argn];
+
+	if(strcmp(arg, "log_level") == 0){
+		set_log_level_run_time = strtoul(argv[++argn], NULL, 10);
```
Since you are storing the result of `strtoul` directly in `set_log_level_run_time` then the type should be `unsigned long`.

00thirdeye00 replied:
changed it to `unsigned long`

Thalley commented on tests/bsim/bluetooth/host/misc/sample_test/src/main.c:
```diff
@@ -9,11 +9,33 @@
+		if(set_log_level_run_time >= 0 && 
+			set_log_level_run_time <= 4){
+			bt_testlib_log_level_set("dut", set_log_level_run_time);
+		}
```
We should abort if an invalid log value is set

00thirdeye00 replied:
can i set to default in that case? setting `LOG_LEVEL_INF` if the `arg` is out of bounds

00thirdeye00 replied:
`TEST_PRINT("Invalid arguments to set log level: %d", runtime_log_level);` printing out as such?

00thirdeye00 replied:
It says device `p_2G4: WARNING: (src/bs_pc_base.c:408): Device 0 left the party unsuspectingly.. I treat it as if it disconnected` when I call `exit(1)` when invalid arguments are received. Is it good?

aescolar replied:
> It says device `p_2G4: WARNING: (src/bs_pc_base.c:408): Device 0 left the party unsuspectingly.. I treat it as if it disconnected` when I call `exit(1)` when invalid arguments are received. Is it good?

The message from the phy (p_2G4) is just telling that the device disappeared from the simulation without properly disconnecting.
You should not call `exit()`. You can instead call, for ex., ```TEST_FAIL("some message");```

Thalley reviewed (changes requested):
Good start, but some suggestions for improvements

theob-pro commented on tests/bsim/bluetooth/host/misc/sample_test/src/main.c:
```diff
@@ -9,11 +9,33 @@
+	if(strcmp(arg, "log_level") == 0){
+		set_log_level_run_time = strtoul(argv[++argn], NULL, 10);
+
+		TEST_PRINT("arg : %d\n", set_log_level_run_time);
```
`TEST_PRINT` already add the ending line return.
```suggestion
		TEST_PRINT("arg : %d", set_log_level_run_time);
```

00thirdeye00 replied:
removed `\n` in `TEST_PRINT`

theob-pro reviewed (changes requested):
This is going the right way. Some details to change.

Also I tried to run it and it's failing, the device 0 doesn't start.

aescolar commented on tests/bsim/bluetooth/host/misc/sample_test/src/main.c:
```diff
@@ -38,6 +60,7 @@ static const struct bst_test_instance entrypoints[] = {
 		.test_id = "dut",
 		.test_delete_f = test_end_cb,
 		.test_main_f = entrypoint_dut,
+		.test_args_f = test_args,
```
@jori-nordic is there any reason why you suggested this being a test argument instead of being registered as a plain command line argument? (so it would be passed as `--log_level=4` instead of  `-argstest log_level 4`. You could have register & handled from the bt testlib directly instead of each test).

Thalley replied:
I was wondering something similar. 
Might be possible to implement this in a more generic way rather than handling it for each test application

aescolar replied:
The native simulator is meant to be really plug&play for this kind of things. So yep, you can do so. Something like adding in that bt testlib library the equivalent of 

https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/native/nrf_bsim/common/phy_sync_ctrl.c#L136
(but registering just one argument)
+ 
A hook like this
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/native/nrf_bsim/common/phy_sync_ctrl.c#L213
to have that command line argument registration function be called very early at runtime.
+
an equivalent to https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/native/nrf_bsim/common/phy_sync_ctrl.c#L100
to handle the argument when/if it is passed to the executable.

jori-nordic replied:
Hi @aescolar it was just OTOH. I guess it's better to do that from the testlib, yeah. With a sensible default ofc.

jori-nordic replied:
To be addressed in a future PR.

00thirdeye00 commented on tests/bsim/bluetooth/host/misc/sample_test/src/main.c:
```diff
@@ -9,11 +9,33 @@
 extern void entrypoint_peer(void);
 extern enum bst_result_t bst_result;
 
+static uint8_t	set_log_level_run_time = 3;
```
this is shorter and much better, I have changed the variable name

00thirdeye00 commented on tests/bsim/bluetooth/host/misc/sample_test/src/dut.c:
```diff
@@ -20,7 +20,7 @@
 #include "data.h"
 
-LOG_MODULE_REGISTER(dut, CONFIG_APP_LOG_LEVEL);
+LOG_MODULE_REGISTER(dut, LOG_LEVEL_DBG);
```
i have now set `peer.c` to `LOG_LEVEL_DBG`

00thirdeye00 replied:
`bt_testlib_log_level_set` this function breaking it

theob-pro replied:
I missed it but by default that test was on log level info. (even if it only matter if log filtering is off as the default log level is info)

```suggestion
LOG_MODULE_REGISTER(dut, LOG_LEVEL_INF);
```

Also, do we want to update `peer.c` too in the end?

Thalley commented on tests/bsim/bluetooth/host/misc/sample_test/src/main.c:
```diff
@@ -9,11 +9,35 @@
 extern void entrypoint_peer(void);
 extern enum bst_result_t bst_result;
 
+static unsigned long runtime_log_level = LOG_LEVEL_INF;
```
This should be a local variable in `test_args`. It is not used outside of it.

00thirdeye00 replied:
My apologies, I will create a fresh PR for this with all review comments. My question is `bt_testlib_log_level_set` this function is breaking it so any suggestions regarding this?

Thalley replied:
> @Thalley My apologies, I will create a fresh PR for this with all review comments. 

Please don't create a new PR :) You can squash the commits instead

> My question is bt_testlib_log_level_set this function is breaking it so any suggestions regarding this?

How do you mean?

00thirdeye00 replied:
> Please don't create a new PR :) You can squash the commits instead
I will do that, thanks.
> How do you mean?
```

d_00: WARNING: (src/bs_pc_base.c:146): Previous lock file found /tmp/bs_combine/tests_bsim_bluetooth_host_misc_sample_test/2G4.d0.lock
d_00: WARNING: (src/bs_pc_base.c:191): Found previous lock owned by DEAD process (pid was 874088), will attempt to take over
d_01: WARNING: (src/bs_pc_base.c:146): Previous lock file found /tmp/bs_combine/tests_bsim_bluetooth_host_misc_sample_test/2G4.d1.lock
d_01: WARNING: (src/bs_pc_base.c:191): Found previous lock owned by DEAD process (pid was 874089), will attempt to take over
p_2G4: WARNING: (src/bs_pc_base.c:146): Previous lock file found /tmp/bs_combine/tests_bsim_bluetooth_host_misc_sample_test/2G4.phy.lock
p_2G4: WARNING: (src/bs_pc_base.c:191): Found previous lock owned by DEAD process (pid was 874090), will attempt to take over
d_00: @00:00:00.000000 INFO: (WEST_TOPDIR/zephyr/tests/bsim/bluetooth/host/misc/sample_test/src/main.c:29): arg : 4

d_00: @00:00:00.000000  <err> [irq] os: Halting system
d_01: @00:00:00.000000  *** Booting Zephyr OS build v3.7.0-rc3-80-g9f25de06cb55 ***
d_01: @00:00:00.000000 INFO: Test start: peer
p_2G4: WARNING: (src/bs_pc_base.c:408): Device 0 left the party unsuspectingly.. I treat it as if it disconnected
timeout: the monitored command dumped core
/home/combine/zephyrproject/zephyr/tests/bsim/sh_common.source: line 22: 874232 Trace/breakpoint trap   $@
timeout: sending signal TERM to command ‘/home/combine/zephyrproject/zephyr/../tools/bsim/bin/bs_nrf52_bsim_tests_bsim_bluetooth_host_misc_sample_test_prj_conf’
timeout: sending signal TERM to command ‘./bs_2G4_phy_v1’
timeout: sending signal KILL to command ‘/home/combine/zephyrproject/zephyr/../tools/bsim/bin/bs_nrf52_bsim_tests_bsim_bluetooth_host_misc_sample_test_prj_conf’
/home/combine/zephyrproject/zephyr/tests/bsim/sh_common.source: line 22: 874233 Killed                  $@

```

theob-pro replied:
You never enable the config for runtime filtering, please read my comment here: https://github.com/zephyrproject-rtos/zephyr/pull/75906#issuecomment-2230036673 and see step 4.

theob-pro replied:
Also, please note that the variable cannot be local. It need to be a global because the run time filtering need to happen in the code of the device itself else, the log module will not be registered.

Thalley replied:
> Also, please note that the variable cannot be local. It need to be a global because the run time filtering need to happen in the code of the device itself else, the log module will not be registered.

Not sure I follow this. 
`unsigned long runtime_log_level = LOG_LEVEL_INF;` is just a integer value used to call `bt_testlib_log_level_set` with as a value (not a reference). It is never used or referenced after that. In a similar fashion it should be possible to even do `bt_testlib_log_level_set("dut", 3);` without any variable at all. 

Can you elaborate on why it would need to be a global or `static` variable?

theob-pro replied:
Yes, sorry. The thing is that you cannot call `bt_testlib_log_level_set` here. You need to call it in the entrypoint function of the device to be able to filter because the log module is declared there.
The variable need to be global, not static.

Thalley replied:
Ah, so the issue is that `bt_testlib_log_level_set` cannot be called in the `test_args_f` function, but needs to done in the `test_main_f` (or some other function in between?). 

In that case, then yes, the shell argument needs to be stored somewhere in the meantime. 
It may be a moot point though as pointed out by https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1681068650 there may be a better way of doing this than using the `test_args_f`

theob-pro replied:
> Ah, so the issue is that `bt_testlib_log_level_set` cannot be called in the `test_args_f` function, but needs to done in the `test_main_f` (or some other function in between?).

It can be called, it will run fine, but it will have no effect on the log level of "dut" in our case.

> It may be a moot point though as pointed out by [#75906 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1681068650) there may be a better way of doing this than using the `test_args_f`

I don't see how it will solve the present issue, `bt_testlib_log_level_set` will still need to be called in the entrypoint function (or somewhere in the actual test code as you said). I feel like it doesn't change a lot of thing.
IMO we should stick with the first proposal and focus on having a first approach working :slightly_smiling_face:

Thalley replied:
> It can be called, it will run fine, but it will have no effect on the log level of "dut" in our case.

I'm clearly missing some information in order to proper review this PR then. 
The logs are registered as `LOG_MODULE_REGISTER(dut, LOG_LEVEL_DBG);`, so the logging module should already be initialized with `LOG_LEVEL_DBG`, so why cannot we call `bt_testlib_log_level_set` (with effect) here? 

The issue is a bit on clear on this as well: Do we want to set the log level per log module, or all of them? 
If the former, then this needs to be handled in `dut.c` as this PR right now only handles "dut". 
If the latter, then this PR needs to gather all the log modules defined in the test files (only), and apply the log level to all of them. 

> IMO we should stick with the first proposal and focus on having a first approach working

I'd rather we come up with a proper design rather than just winging it :D 

I think there are some fundamental things we need to agree on before going forwards such as

- Whether to apply the log level to just a single logging module or multiple (a single test may define multiple log modules)
- Whether to have individual command line handler per source file (as this PR somewhat does), or handle it more generically in the bt_testlib or the babblekit (as this is not really BT related). 
- Whether to handle it in the test_args_f or as a command line arg as suggested by https://github.com/zephyrproject-rtos/zephyr/pull/75906/files#r1681068650

theob-pro replied:
> I'm clearly missing some information in order to proper review this PR then. The logs are registered as `LOG_MODULE_REGISTER(dut, LOG_LEVEL_DBG);`, so the logging module should already be initialized with `LOG_LEVEL_DBG`, so why cannot we call `bt_testlib_log_level_set` (with effect) here?

The log module is registered in Zephyr but when `test_args` is run it's within BabbleSim, the device haven't booted yet.

> Do we want to set the log level per log module, or all of them?

Again, I really think that is a detail. We are in a sample test that can evolve or be updated later. We could do complex stuff like having a pair of argument, one that specify the module and the other that give the log level. But that's more complex and I think it's good to start small ^^

Thalley reviewed (changes requested):
Please read https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow on how to update your PR based on review comments (step 14)

00thirdeye00 committed:
tests: bsim: set log level from command-line argument

Implementation to take command-line arguments to set log level in run
time for bsim module. Set default log level if no log level is passed
at runtime

Signed-off-by: Guru Mehar Rachaputi <gurumeharrachaputi@gmail.com>

theob-pro commented:
@00thirdeye00 as your solution is getting better, it's still not working properly. Please read the discussion over there https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1683794045. Also, you have compliance failure and some `TEST_PRINT` still have the line return.

00thirdeye00 commented:
> @00thirdeye00 as your solution is getting better, it's still not working properly. Please read the discussion over there [#75906 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1683794045). Also, you have compliance failure and some `TEST_PRINT` still have the line return.

could you confirm if [this](https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1686973060) is ok

theob-pro commented:
> > @00thirdeye00 as your solution is getting better, it's still not working properly. Please read the discussion over there [#75906 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1683794045). Also, you have compliance failure and some `TEST_PRINT` still have the line return.
> 
> Hej @theob-pro, I believe I have updated as per the discussion. Could you check and also confirm [this](https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1686973060) is ok

It's working now, thanks.
For the exit you should rather make the test fail. You can use the `TEST_FAIL` macro.

Also, about that discussion https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1681068650 do you have an opinion? It seems most people involved here would prefer to have that.

00thirdeye00 commented:
> > > @00thirdeye00 as your solution is getting better, it's still not working properly. Please read the discussion over there [#75906 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1683794045). Also, you have compliance failure and some `TEST_PRINT` still have the line return.
> > 
> > 
> > Hej @theob-pro, I believe I have updated as per the discussion. Could you check and also confirm [this](https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1686973060) is ok
> 
> It's working now, thanks. For the exit you should rather make the test fail. You can use the `TEST_FAIL` macro.
> 
> Also, about that discussion [#75906 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1681068650) do you have an opinion? It seems most people involved here would prefer to have that.
All the tests under bsim seems to follow `-argstest log_level 4` test argument style. Maybe its good to keep the same pattern.

theob-pro commented on tests/bsim/bluetooth/host/misc/sample_test/src/dut.c:
```diff
@@ -98,6 +101,9 @@ void entrypoint_dut(void)
 	/* Initialize device sync library */
 	bk_sync_init();
 
+	/* Runtime log filtering */
+	bt_testlib_log_level_set("dut", runtime_log_level);
```
```suggestion
	/* Set the log level given by the `log_level` CLI argument */
	bt_testlib_log_level_set("dut", runtime_log_level);
```

theob-pro commented on tests/bsim/bluetooth/host/misc/sample_test/src/main.c:
```diff
@@ -9,11 +9,33 @@
+			runtime_log_level <= LOG_LEVEL_DBG){
+			TEST_PRINT("Runtime log level configuration: %d", runtime_log_level);
+		} else {
+			TEST_FAIL("Invalid arguments to set log level: %d\n", runtime_log_level);
```
```suggestion
			TEST_FAIL("Invalid arguments to set log level: %d", runtime_log_level);
```

00thirdeye00 replied:
I have update peer module log registration to `LOG_MODULE_REGISTER(peer, LOG_LEVEL_INF)` now.

00thirdeye00 replied:
@theob-pro thanks for the comments, updated the PR now :)

theob-pro replied:
Great, but regarding `peer.c` you just hard coded the log level, why not use the run time filtering like in the DUT?

00thirdeye00 replied:
@theob-pro I just want to clarify if the same `log_level` is set to `peer` module as well.

00thirdeye00 replied:
@theob-pro, When I pass runtime arg bigger than level the module is registered, I get the error. Is it expected? Below is the error:
```
d_00: WARNING: (src/bs_pc_base.c:146): Previous lock file found /tmp/bs_combine/tests_bsim_bluetooth_host_misc_sample_test/2G4.d0.lock
d_00: WARNING: (src/bs_pc_base.c:191): Found previous lock owned by DEAD process (pid was 962583), will attempt to take over
d_00: @00:00:00.000000 INFO: (WEST_TOPDIR/zephyr/tests/bsim/bluetooth/host/misc/sample_test/src/main.c:31): Runtime log level configuration: 4
d_01: @00:00:00.000000 INFO: (WEST_TOPDIR/zephyr/tests/bsim/bluetooth/host/misc/sample_test/src/main.c:31): Runtime log level configuration: 4
d_00: @00:00:00.000000  *** Booting Zephyr OS build v3.7.0-rc3-109-g82ac3a7eb0e1 ***
d_00: @00:00:00.000000 INFO: Test start: dut
d_01: @00:00:00.000000  *** Booting Zephyr OS build v3.7.0-rc3-109-g82ac3a7eb0e1 ***
d_01: @00:00:00.000000 INFO: Test start: peer
d_00: @00:00:00.000000  ASSERTION FAIL [result_level == new_level] @ WEST_TOPDIR/zephyr/tests/bluetooth/common/testlib/include/testlib/log_utils.h:22
d_01: @00:00:00.000000  ASSERTION FAIL [result_level == new_level] @ WEST_TOPDIR/zephyr/tests/bluetooth/common/testlib/include/testlib/log_utils.h:22
d_00: @00:00:00.000000  	3 4
d_01: @00:00:00.000000  	3 4
d_01: @00:00:00.000000  @ WEST_TOPDIR/zephyr/lib/os/assert.c:43
d_00: @00:00:00.000000  @ WEST_TOPDIR/zephyr/lib/os/assert.c:43
d_00: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
d_01: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
d_00: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: Current thread: 0x825dbc0 (main)
d_00: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: Halting system
d_01: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: Current thread: 0x825dbc0 (main)
d_01: @00:00:00.000000  [00:00:00.000,000] <err> [main] os: Halting system
p_2G4: WARNING: (src/bs_pc_base.c:408): Device 0 left the party unsuspectingly.. I treat it as if it disconnected
timeout: the monitored command dumped core
/home/combine/zephyrproject/zephyr/tests/bsim/sh_common.source: line 22: 963019 Trace/breakpoint trap   $@
p_2G4: WARNING: (src/bs_pc_base.c:408): Device 1 left the party unsuspectingly.. I treat it as if it disconnected
timeout: the monitored command dumped core
/home/combine/zephyrproject/zephyr/tests/bsim/sh_common.source: line 22: 963020 Trace/breakpoint trap   $@
```
Is it a good idea to register `peer` and `dut` modules with `LOG_LEVEL_DBG`?

theob-pro replied:
My bad, the thing is you cannot set a level higher that the default of level of the module. So you were right to do `LOG_MODULE_REGISTER(..., LOG_LEVEL_DBG)` as this is the highest level available.

00thirdeye00 replied:
@theob-pro thank for the confirmation. I have updated it now.

theob-pro replied:
Great, one last thing, you could remove the Kconfig now that it's not use anymore. (also remove the commented config in the project config)

00thirdeye00 replied:
Hej @theob-pro , I have updated now

theob-pro replied:
Just remove `CONFIG_APP_LOG_LEVEL_DBG` in the comment, the others are still useful.

theob-pro reviewed (commented):
> > Also, about that discussion [#75906 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/75906#discussion_r1681068650) do you have an opinion? It seems most people involved here would prefer to have that.
> 
> All the tests under bsim seems to follow `-argstest log_level 4` test argument style. Maybe its good to keep the same pattern.

Seems reasonable to me. Also we can always update that later.

The PR is working as expected now. Just some small changes.

jori-nordic commented on tests/bsim/bluetooth/host/misc/sample_test/test_scripts/run.sh:
```diff
@@ -62,8 +62,10 @@ cd ${BSIM_OUT_PATH}/bin
-Execute "${test_exe}" -v=${verbosity_level} -s=${simulation_id} -d=1 -rs=69  -testid=peer
+Execute "${test_exe}" -v=${verbosity_level} -s=${simulation_id} -d=0 -rs=420 -testid=dut \
+		-argstest log_level 3
+Execute "${test_exe}" -v=${verbosity_level} -s=${simulation_id} -d=1 -rs=69  -testid=peer \
```
We could set this from a shell variable declared at the top. E.g. `-argstest log_level ${test_log_level}`. Can be done in a future PR :+1:

jori-nordic reviewed (approved):
@00thirdeye00 you need to update the commit message title, it's currently too long. 
Re-using the PR title should do it:
`tests: bsim: set log level from command-line argument`

Thalley reviewed (approved):
While I think there are some ways that we perhaps can make this more generic (maybe even put the parsing of the argument into the babblekit), this PR is now showcasing a working minimal implementation. We can improve it from here :)

00thirdeye00 commented:
@jori-nordic i want to confirm, how writing same commit message again works?

jori-nordic commented:
Sorry, my message was misleading. The failing check is apparently in the body.
Anyway, it's easily accessible, right below your comment.

To prevent it, you can reflow the text to columns everywhere, body and commit message. See also https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines.
You can also run the [`scripts/ci/check_compliance.py`](https://docs.zephyrproject.org/latest/contribute/guidelines.html#check-compliance-py) locally for faster feedback.

![image](https://github.com/user-attachments/assets/bb5baf45-0364-4615-8c95-dcd60fc7b08a)

![image](https://github.com/user-attachments/assets/7366e1e3-e1e1-41dc-9231-579bf4a2e753)

00thirdeye00 commented:
@jori-nordic thanks for the clarification, I have made the changes now.

kartben commented:
rebased to address unrelated ci failur
